website/content/docs/concepts/plugins/storage/host-volumes.mdx

```diff
@@ -0,0 +1,387 @@
+    stderr "creating directory: $path"
+    # `mkdir -p` may be run repeatedly with the same result;
+    # it is idempotent.
+    mkdir -p "$path"
+    # 0 bytes because plain directories are not any particular size.
+    printf '{"path": "%s", "bytes": 0}' "$path"
+    ;;
+  "delete")
+    path="$(get_path)"
+    stderr "deleting directory: $path"
+    rm -rf "$path" # `rm -f` is also idempotent.
+    ;;
+  *)
+    echo "unknown operation: '$DHV_OPERATION'"
+    exit 1
+    ;;
+esac
+```
+
+</Tab>
+<Tab heading="mkfs-ext4">
+
+`mkfs-ext4` creates a Linux ext4 filesystem with the `mkfs.ext4` command and
+mounts it  as a loopback device. Unlike `mkdir`, `mkfs` can restrict the size
+of the volume.
+
+Volume specification:
+```hcl
+type         = "host"
+name         = "mkfs-vol"
+plugin_id    = "mkfs-ext4" # plugin filename
+capacity_min = "50MB" # capacity values are required by this plugin
+capacity_max = "50MB"
+```
+
+Plugin code:
+
+```bash
+#!/usr/bin/env bash
+set -euo pipefail
+
+version='0.0.1'
+
+stderr() {
+  1>&2 echo "$@"
+}
+
+get_path() {
+  if [ -z "$DHV_HOST_PATH" ]; then
+    stderr "DHV_HOST_PATH must not be empty"
+    exit 1
+  fi
+  if [ -z "$DHV_VOLUME_NAME" ]; then
+    stderr "DHV_VOLUME_NAME must not be empty"
+    exit 1
+  fi
+  echo "$DHV_HOST_PATH/$DHV_VOLUME_NAME"
+}
+
+is_mounted() {
+  mount | grep -q " $1 "
+}
+
+create_volume() {
+    local path="$1"
+    local bytes="$2"
+
+    # translate to mb for dd block size
+    local megs=$((bytes / 1024 / 1024)) # lazy, approximate
+    if [ $megs -le 0 ]; then
+      stderr "minimum capacity must be greater than zero."
+      exit 2
+    fi
+
+    mkdir -p "$path"
+    # the if statements ensure idempotency
+    if [ ! -f "$path.ext4" ]; then
+      # dd only writes to stderr, so safe to run without redirection
+      dd if=/dev/zero of="$path.ext4" bs=1M count="$megs"
+      # mkfs includes stdout, so we need to redirect that to stderr
+      mkfs.ext4 "$path.ext4" 1>&2
+    fi
+    if ! is_mounted "$path"; then
+      mount "$path.ext4" "$path"
+    fi
+}
+
+delete_volume() {
+  local path="$1"
+  is_mounted "$path" && umount "$path"
+  rm -rf "$path"
+  rm -f "$path.ext4"
+}
+
+case "$1" in
+  "fingerprint")
+    printf '{"version": "%s"}' "$version"
+    ;;
+  "create")
+    path="$(get_path)"
+    stderr "creating volume at $path"
+    create_volume "$path" "$DHV_CAPACITY_MIN_BYTES"
+    # output what Nomad expects
+    bytes="$(stat --format='%s' "$path.ext4")"
+    printf '{"path": "%s", "bytes": %s}' "$path" "$bytes"
+    ;;
+  "delete")
+    path="$(get_path)"
+    stderr "deleting volume at $path"
+    delete_volume "$path" ;;
+  *)
+    echo "unknown operation: $1"
+    exit 1 ;;
+esac
+```
+
+</Tab>
+</Tabs>
+
+## Specification
+
+A plugin is registered with Nomad if it:
+
+* Is an executable file (a script or binary)
+* Is located in the [`client.host_volume_plugin_dir`][plugin_dir] directory
+* Responds appropriately to a `fingerprint` call
```

Comment:
Just to make it unmistakable, maybe we should note that the plugins need to be installed on clients and not servers. That's "obvious", and certainly implies by the config value, but...